Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -73,140 +74,186 @@ class WalletConnectManager {
             guard let safe = try? Safe.getSelected() else { return }
+            
             var sessionNamespaces = [String: SessionNamespace]()
             proposal.requiredNamespaces.forEach {
                 let caip2Namespace = $0.key
                 let proposalNamespace = $0.value
-                let chains = proposalNamespace.chains
-
-                let accounts = Set(chains.compactMap {
+                guard let chains = proposalNamespace.chains else { return }
+                
+                let selectedSafeChain = chains.filter { chain in
+                    chain.namespace == EVM_COMPATIBLE_NETWORK && chain.reference == safe.chain?.id
+                }
+                
+                let accounts = Set(selectedSafeChain.compactMap {
                     Account($0.absoluteString + ":\(safe.addressValue)")
                 })
-
+                
                 let sessionNamespace = SessionNamespace(accounts: accounts,
                                                         methods: proposalNamespace.methods,
                                                         events: proposalNamespace.events)
                 sessionNamespaces[caip2Namespace] = sessionNamespace
             }
             do {
-                try await Sign.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
+                try await Web3Wallet.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
             } catch {
                 print("DAPP: Approve Session error: \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(error: GSError.error(description: "Approve Session error: ", error: error))
+                }
             }
         }
     }
-
+    
     /// By default, session lifetime is set for 7 days and after that time user's session will expire.
     /// This method will extend the session for 7 days
     func extend(session: Session) async {
         do {
             try await Web3Wallet.instance.extend(topic: session.topic)
         } catch {
+            Task { @MainActor in
+                App.shared.snackbar.show(error: GSError.error(description: error.localizedDescription, error: error))
+            }
             print("DAPP: extending Session error: \(error)")
         }
     }
-
+    
     func disconnect(session: Session) {
         Task {
             do {
                 NotificationCenter.default.post(name: .wcDidDisconnectSafeServer, object: self)
-                try await Sign.instance.disconnect(topic: session.topic)
+                try await Web3Wallet.instance.disconnect(topic: session.topic)
             } catch {
-                print("DAPP: disconnectting Session error: \(error)")
+                print("DAPP: disconnecting Session error: \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(error: GSError.error(description: "Disconnecting Session error", error: error))
```

Comment:
Please Create new GSError struct in 'GSError' file and use it instead

Reply:
I added for new DetailedLocalizedError. Two generic errors and two specific 
1. pairing failed because it already existed
2. session creation failed because a non-safe chanin was in the required namespaces